### PR TITLE
MetaDataRef: Add contains() and get()

### PR DESCRIPTION
**Pull request by rubenwardy**
Both of these allow for nicer code

### CODE REVIEW

File: src/script/lua_api/l_metadata.cpp [hunk 1]
```diff
@@ -51,6 +51,46 @@ MetaDataRef* MetaDataRef::checkobject(lua_State *L, int narg)
...
+int MetaDataRef::l_get_string_or_nil(lua_State *L)
+{
+	MAP_LOCK_REQUIRED;
+
+	MetaDataRef *ref = checkobject(L, 1);
+	std::string name = luaL_checkstring(L, 2);
+
+	Metadata *meta = ref->getmeta(false);
+	if (meta == NULL) {
+		lua_pushlstring(L, "", 0);
```
Comment (sfan5) [inline src/script/lua_api/l_metadata.cpp]: :thinking:

shouldn't this just throw an error or something?

Comment (nerzhul) [reply src/script/lua_api/l_metadata.cpp]: return 0 ?

Comment (SmallJoker) [reply src/script/lua_api/l_metadata.cpp]: @nerzhul That's the same, but yes.

**Comment by rubenwardy** [comment]
Updated. Changed to `get()` in order to be more consistent with the Settings API. Can't fix the `set_type`s however :(

File: src/script/lua_api/l_metadata.cpp [hunk 2]
```diff
@@ -51,6 +51,39 @@ MetaDataRef* MetaDataRef::checkobject(lua_State *L, int narg)
...
+// contains(self, name)
+int MetaDataRef::l_contains(lua_State *L)
+{
+	MAP_LOCK_REQUIRED;
+
+	MetaDataRef *ref = checkobject(L, 1);
+	std::string name = luaL_checkstring(L, 2);
+
+	Metadata *meta = ref->getmeta(false);
+	lua_pushboolean(L, meta != NULL && meta->contains(name));
```
Comment (sfan5) [inline src/script/lua_api/l_metadata.cpp]: shouldn't this also `return 0` is meta is NULL?

**Pull request merged**